pkg/lib/indexconfig/indexconfig_test.go

```diff
@@ -0,0 +1,108 @@
+package indexconfig
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/operator-framework/operator-registry/pkg/registry"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	manifestDir      = "./testdata/bundle/manifests"
+	tmpTestDirPrefix = "tmp_test_"
+	expectedConfig   = "{\n    \"schema\": \"olm.package\",\n    \"name\": \"lib-bucket-provisioner\",\n    \"defaultChannel\": \"alpha\",\n    \"icon\": {\n       \"base64data\": \"\",\n       \"mediatype\": \"\"\n    },\n    \"channels\": [\n       \"alpha\",\n       \"beta\"\n    ],\n    \"description\": \"\"\n }\n{\n    \"schema\": \"olm.bundle\",\n    \"name\": \"lib-bucket-provisioner.v1.0.0\",\n    \"package\": \"lib-bucket-provisioner\",\n    \"image\": \"\",\n    \"version\": \"1.0.0\",\n    \"properties\": [\n       {\n          \"type\": \"olm.channel\",\n          \"value\": {\n             \"name\": \"alpha\",\n             \"replaces\": \"\"\n          }\n       },\n       {\n          \"type\": \"olm.channel\",\n          \"value\": {\n             \"name\": \"beta\",\n             \"replaces\": \"\"\n          }\n       },\n       {\n          \"type\": \"skips\",\n          \"value\": \"\"\n       },\n       {\n          \"type\": \"skipsRange\",\n          \"value\": \"\"\n       }\n    ],\n    \"relatedImages\": []\n }\n"
+)
+
+func TestPrintConfigCreatesNewConfig(t *testing.T) {
+
+	// create bundle from manifests
+	bundle, err := testBundle()
+	require.NoError(t, err)
+	// create temporary dir for test
+	tmpDir, err := ioutil.TempDir("./", tmpTestDirPrefix)
+	defer os.RemoveAll(tmpDir)
+	require.NoError(t, err)
```

Comment:
This seems like something that abstracted away a bit as a test setup helper function, where you could pass a testing.T parameter and uses that new `t.Cleanup` functionality that was added in 1.14.

Reply:
So I'm on 1.13, and I'm wondering if just `t.Cleanup` is enough justification to tell a system on 1.13 you can't run the tests unless you upgrade to 1.14.

Reply:
FWIW, Go 1.16 is out now, so Go 1.13 is pretty out of date. On many of the projects I've worked on we've bumped Go versions soon after each release and I haven't heard contributors complain. Go is _super_ stable and rarely causes problems. Also, I'm fairly sure Go 1.15 is being used to build OLM's packages for OpenShift builds, so we may as well use at least that.